shell/components/dialog/ConfirmNameToRemoveDialog.vue

```diff
@@ -0,0 +1,108 @@
+<script>
+import { mapGetters } from 'vuex';
+import { exceptionToErrorsArray } from '@shell/utils/error';
+import { alternateLabel } from '@shell/utils/platform';
+import AsyncButton from '@shell/components/AsyncButton';
+import { Banner } from '@components/Banner';
+import { Card } from '@components/Card';
+import CopyToClipboardText from '@shell/components/CopyToClipboardText';
+
+/**
+ * @name ConfirmNameToRemoveDialog
+ * @description Dialog component to confirm name before removing the resource.
+ */
+export default {
+  name: 'ConfirmNameToRemoveDialog',
+
+  components: {
+    AsyncButton,
+    Banner,
+    Card,
+    CopyToClipboardText
+  },
+
+  props:      {
+    resources: {
+      type:     Array,
+      required: true
+    }
+  },
+
+  data() {
+    return { errors: [], confirmName: '' };
+  },
+
+  computed: {
+    ...mapGetters({ t: 'i18n/t' }),
+
+    nameToMatch() {
+      return this.resources[0].nameDisplay;
+    },
+
+    deleteDisabled() {
+      const confirmFailed = this.confirmName !== this.nameToMatch;
```

Comment:
Do we make strict checks about cases and spaces or are we more flexible? Not sure what's the typical approach we have.